[Core] `bazel build -c fastbuild //:ray_pkg` fails on linux

### What happened + What you expected to happen

I was following the [instructions for building ray from source](https://docs.ray.io/en/latest/ray-contribute/development.html#fast-debug-and-optimized-builds); I ran `bazel build -c fastbuild //:ray_pkg` at the root of the repository. The build nearly finishes but before it does, I get an error about an uninitialized variable:

```
ERROR: /home/pdmurray/Desktop/workspace/ray/BUILD.bazel:782:11: Compiling src/ray/core_worker/gcs_server_address_updater.cc failed: (Exit 1): gcc failed: error executing command

...(huge gcc command removed here for brevity)

src/ray/core_worker/gcs_server_address_updater.cc: In constructor 'ray::core::GcsServerAddressUpdater::GcsServerAddressUpdater(std::string, int, std::function<void(std::__cxx11::basic_string<char>, int)>)':
src/ray/core_worker/gcs_server_address_updater.cc:24:28: error: member 'ray::core::GcsServerAddressUpdater::updater_io_service_' is used uninitialized [-Werror=uninitialized]
   24 |     : client_call_manager_(updater_io_service_),
      |                            ^~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
Target //:ray_pkg failed to build
INFO: Elapsed time: 875.684s, Critical Path: 164.63s
INFO: 2795 processes: 1111 internal, 1684 linux-sandbox.
FAILED: Build did NOT complete successfully
```

It looks like `updater_io_service_` is not initialized before it is used in the constructor for `GcsServerAddressUpdater`. `updater_io_service_` contains state needed to talk to the system's io services; should the `GcsServerAddressUpdater` create a new `instrumented_io_context` instance, or should it be retrieved from somewhere else upon instantiation?

I'd be happy to submit a PR to correct this, if the maintainers would accept it.

### Versions / Dependencies

Bazel 5.1.1
Ray (`master` branch and `v1.12.0` tag were tried with the same result)
Python 3.9.12
`uname -rms`: Linux 5.17.7-arch1-2 x86_64
GCC 12.1.0

### Reproduction script

```bash
#!/bin/bash
bazel build -c fastbuild //:ray_pkg
```

### Issue Severity

High: It blocks me from completing my task.

--- a/src/ray/core_worker/gcs_server_address_updater.h
+++ b/src/ray/core_worker/gcs_server_address_updater.h
@@ -38,11 +38,11 @@ class GcsServerAddressUpdater {
   /// Update gcs server address.
   void UpdateGcsServerAddress();
 
+  instrumented_io_context updater_io_service_;
   rpc::ClientCallManager client_call_manager_;
   /// A client connection to the raylet.
   raylet::RayletClient raylet_client_;
   std::function<void(std::string, int)> update_func_;
-  instrumented_io_context updater_io_service_;
   PeriodicalRunner updater_runner_;
   std::thread updater_thread_;
   int32_t failed_ping_count_ = 0;

